src/Nox.Solution/NoxSolution.cs

```diff
@@ -39,6 +39,53 @@ internal void Validate()
         return fullRelationshipModels;
     }
 
+    public List<EntityRelationshipWithType> GetOwnedRelationshipsToCreate(
+        Func<string, Type?> getTypeByNameFunc,
+        Entity entity)
+    {
+        var fullRelationshipModels = new List<EntityRelationshipWithType>();
+
+        if (entity?.OwnedRelationships != null)
+        {
+            foreach (var relationship in entity.OwnedRelationships)
+            {
+                fullRelationshipModels.Add(new EntityRelationshipWithType
+                {
+                    Relationship = relationship,
+                    RelationshipEntityType = getTypeByNameFunc(relationship.Entity)!
+                });
+            }
+        }
+
+        return fullRelationshipModels;
+    }
+
+    public bool IsOwnedEntity(Entity entity)
+    {
+        if (Domain == null)
+        {
+            return false;
+        }
+
+        foreach(var domainEntity in Domain.Entities)
```

Comment:
improve performance by  having a hashset of OwnedRelationships Entity name, and the use it for search

the hashset can be a input of the method calculate in EvaluateCalculatedProperties a single time (no local state is needed)